aztk/spark/client/client.py

```diff
@@ -0,0 +1,232 @@
+
+import aztk
+from aztk import error
+from aztk import models as base_models
+from aztk.client import CoreClient
+from aztk.internal.cluster_data import NodeData
+from aztk.spark import models
+from aztk.spark.client.cluster import ClusterOperations
+from aztk.spark.client.job import JobOperations
+from aztk.spark.helpers import cluster_diagnostic_helper
+from aztk.spark.helpers import create_cluster as create_cluster_helper
+from aztk.spark.helpers import get_log as get_log_helper
+from aztk.spark.helpers import job_submission as job_submit_helper
+from aztk.spark.helpers import submit as cluster_submit_helper
+from aztk.spark.utils import util
+from aztk.utils import azure_api, deprecated, helpers
+
+
+class Client(CoreClient):
+    """The client used to create and manage Spark clusters
+
+        Attributes:
+            cluster (:obj:`aztk.spark.client.cluster.ClusterOperations`): Cluster
+            job (:obj:`aztk.spark.client.job.JobOperations`): Job
+    """
+    def __init__(self, secrets_configuration: models.SecretsConfiguration = None, **kwargs):
+        self.secrets_configuration = None
+        context = None
+        if kwargs.get("secrets_config"):
+            # TODO: add deprecated warning
+            context = self._get_context(kwargs.get("secrets_config"))
+        else:
+            context = self._get_context(secrets_configuration)
+        self.cluster = ClusterOperations(context)
+        self.job = JobOperations(context)
+
+
+    # ALL THE FOLLOWING METHODS ARE DEPRECATED AND WILL BE REMOVED IN 0.10.0
+
+    @deprecated("0.10.0")
+    def create_cluster(self, cluster_conf: models.ClusterConfiguration, wait: bool = False):
+        return self.cluster.create(cluster_configuration=cluster_conf, wait=wait)
+
+    @deprecated("0.10.0")
+    def create_clusters_in_parallel(self, cluster_confs):    # NOT IMPLEMENTED
+        for cluster_conf in cluster_confs:
+            self.cluster.create(cluster_conf)
+
+    @deprecated("0.10.0")
+    def delete_cluster(self, cluster_id: str, keep_logs: bool = False):
+        return self.cluster.delete(id=cluster_id, keep_logs=keep_logs)
+
+    @deprecated("0.10.0")
+    def get_cluster(self, cluster_id: str):
+        return self.cluster.get(id=cluster_id)
+
+    @deprecated("0.10.0")
+    def list_clusters(self):
+        return self.cluster.list()
+
+    @deprecated("0.10.0")
+    def get_remote_login_settings(self, cluster_id: str, node_id: str):
+        return self.cluster.get_remote_login_settings(cluster_id, node_id)
+
+    @deprecated("0.10.0")
+    def submit(self,
+               cluster_id: str,
+               application: models.ApplicationConfiguration,
+               remote: bool = False,
+               wait: bool = False):
+        return self.cluster.submit(id=cluster_id, application=application, remote=remote, wait=wait)
+
+    @deprecated("0.10.0")
+    def submit_all_applications(self, cluster_id: str, applications):    # NOT IMPLEMENTED
```

Comment:
# NOT IMPLEMENTED?

Reply:
This is just a reminder to us that this function doesn't exist in the new API (i.e. I didn't just forget to do it, it's intentionally not there).